public/app/features/explore/TraceView/createSpanLink.tsx

```diff
@@ -148,18 +152,31 @@ function getLokiQueryFromSpan(span: TraceSpan, options: TraceToLogsOptions): str
     return acc;
   }, [] as string[]);
 
-  // If no tags found, return undefined to prevent an invalid Loki query
-  if (!tags.length) {
+  /** If no tags are found and it's a Loki query, return undefined to prevent
+   * an invalid Loki query. However tags arent required for splunk queries.
+   */
+  if (!tags.length && !isSplunkDS) {
     return undefined;
   }
 
-  let query = `{${tags.join(', ')}}`;
+  let query = '';
+  if (tags.length > 0) {
+    if (!isSplunkDS) {
+      query += `{${tags.join(', ')}}`;
+    } else {
+      query += `${tags.join(' ')}`;
+    }
+  }
 
-  if (filterByTraceID && span.traceID) {
+  if (filterByTraceID && span.traceID && !isSplunkDS) {
     query += ` |="${span.traceID}"`;
+  } else if (filterByTraceID && span.traceID && isSplunkDS) {
+    query += ` TraceID=${span.traceID}`;
```

Comment:
I'd confirm with @toddtreece, but I think we should just search for the raw id instead of the KV pair. If the key will always be 'TraceID', then this is okay, otherwise we need to make the key configurable in the TraceToLogsSettings. 

Same goes for SpanID.

Reply:
it looks like splunk will search the `_raw` field if only the value is specified. searching by value seems like a fine place to start as a MVP to me, and the option to specify a key could be added in a later release if there's a need